### PR TITLE
Changed everything to use eframe instead of bare egui

### PR DESCRIPTION
**Pull request by newcomb-luke**
This switches us from using bare egui and having to do the event loop and everything ourselves, to using eframe, which is egui's "framework".

This does cause us to switch backends from glium to glow, but that shouldn't be an issue.

All functionality from the past version is there.

### CODE REVIEW

File: src/dial_widget.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    + DIAL_ANGLE_OFFSET;
+                let end_radians = (self.in_range.end / DIAL_MAX_VALUE) * std::f32::consts::TAU
+                    + DIAL_ANGLE_OFFSET;
+                let radians_dist = (end_radians - start_radians) / 100.0;
+                let start_x = (radius + DIAL_BAR_WIDTH * 1.0) * f32::cos(start_radians);
+                let start_y = (radius + DIAL_BAR_WIDTH * 1.0) * f32::sin(start_radians);
+
+                let mut last_vertex_pos = Pos2::new(center.x + start_x, center.y + start_y);
+
+                for i in 1..101 {
```
Comment (TroyNeubauer) [inline src/dial_widget.rs]: what is this? Because of the angles would `for i in 0..100` just work?

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,7 @@
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use dials_research::run;
```
Comment (TroyNeubauer) [inline src/main.rs]: would prefer just `dials_research::run()`, the import seems a little wierd

**Review by TroyNeubauer — APPROVED** [review]
other than some little nits, looks good

File: src/main.rs (hunk 2 above)
Comment (newcomb-luke) [reply src/main.rs]: Yes

**Pull request merged**